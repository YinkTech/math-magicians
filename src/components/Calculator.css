
.btnColor {
  color: #555;
  width: 50px;
  font-size: 20px;
}
