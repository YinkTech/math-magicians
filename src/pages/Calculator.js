import React from 'react'
import App from '../components/App'

const Calculator = () => {
  return (
    <div className='calculator'>
      <h3>Letâ€™s do some math!</h3>
      <App />
    </div>
  )
}

export default Calculator;