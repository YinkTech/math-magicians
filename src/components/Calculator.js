import React from 'react';
import './Calculator.css';

class Calculator extends React.Component {
    constructor(props) {
        super(props);
        this.state= {answer: 0};
    };

    render() {
        const { answer } = this.state;
        return(
            <div>
                <p> {answer} </p>
                <div>
                    <div>
                        <button type='button' className='btnColor'>AC</button>
                        <button type='button' className='btnColor'>+/-</button>
                        <button type='button' className='btnColor'>%</button>
                        <button type='button' className='btnColor'>/</button>
                    </div>
                    <div>
                        <button type='button' className='btnColor'>7</button>
                        <button type='button' className='btnColor'>8</button>
                        <button type='button' className='btnColor'>9</button>
                        <button type='button' className='btnColor'>*</button>
                    </div>
                    <div>
                        <button type='button' className='btnColor'>4</button>
                        <button type='button' className='btnColor'>5</button>
                        <button type='button' className='btnColor'>6</button>
                        <button type='button' className='btnColor'>-</button>
                    </div>
                    <div>
                        <button type='button' className='btnColor'>1</button>
                        <button type='button' className='btnColor'>2</button>
                        <button type='button' className='btnColor'>3</button>
                        <button type='button' className='btnColor'>+</button>
                    </div>
                    <div>
                        <button type='button' className='btnColor'>0</button>
                        <button type='button' className='btnColor'>.</button>
                        <button type='button' className='btnColor'>=</button>
                    </div>
                </div>

                
            </div>
        )
    }
}

export default Calculator;