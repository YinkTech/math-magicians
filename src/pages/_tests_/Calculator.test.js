import React from "react";
import renderer from "react-test-renderer";
import Calculator from "./../Calculator";

it("If it renders correctly", () => {
    const cal = renderer
    .create(<Calculator />)
    .toJSON();
    expect(cal).toMatchSnapshot();
});