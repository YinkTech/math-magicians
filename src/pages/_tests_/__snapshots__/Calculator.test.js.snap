// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`If it renders correctly 1`] = `
<div
  className="calculator"
>
  <h1>
    Let’s do some math!
  </h1>
  <div
    className="calBody"
  >
    <div
      className="display"
    >
      <span>
         
        0
         
      </span>
      <span>
         
         
      </span>
      <span>
         
         
      </span>
    </div>
    <div>
      <div>
        <button
          className="btnName"
          name="AC"
          onClick={[Function]}
          type="button"
        >
           
          AC
        </button>
        <button
          className="btnName"
          name="+/-"
          onClick={[Function]}
          type="button"
        >
           
          +/-
        </button>
        <button
          className="btnName"
          name="%"
          onClick={[Function]}
          type="button"
        >
           
          %
        </button>
        <button
          className="btnName cal"
          name="÷"
          onClick={[Function]}
          type="button"
        >
           
          ÷
        </button>
      </div>
      <div>
        <button
          className="btnName number"
          name="7"
          onClick={[Function]}
          type="button"
        >
           
          7
        </button>
        <button
          className="btnName number"
          name="8"
          onClick={[Function]}
          type="button"
        >
           
          8
        </button>
        <button
          className="btnName number"
          name="9"
          onClick={[Function]}
          type="button"
        >
           
          9
        </button>
        <button
          className="btnName cal cal-sec"
          name="X"
          onClick={[Function]}
          type="button"
        >
           
          X
        </button>
      </div>
      <div>
        <button
          className="btnName number"
          name="4"
          onClick={[Function]}
          type="button"
        >
           
          4
        </button>
        <button
          className="btnName number"
          name="5"
          onClick={[Function]}
          type="button"
        >
           
          5
        </button>
        <button
          className="btnName number"
          name="6"
          onClick={[Function]}
          type="button"
        >
           
          6
        </button>
        <button
          className="btnName cal cal-sec"
          name="-"
          onClick={[Function]}
          type="button"
        >
           
          -
        </button>
      </div>
      <div>
        <button
          className="btnName number"
          name="1"
          onClick={[Function]}
          type="button"
        >
           
          1
        </button>
        <button
          className="btnName number"
          name="2"
          onClick={[Function]}
          type="button"
        >
           
          2
        </button>
        <button
          className="btnName number"
          name="3"
          onClick={[Function]}
          type="button"
        >
           
          3
        </button>
        <button
          className="btnName cal"
          name="+"
          onClick={[Function]}
          type="button"
        >
           
          +
        </button>
      </div>
      <div>
        <button
          className="btnName last zero number"
          name="0"
          onClick={[Function]}
          type="button"
        >
           
          0
        </button>
        <button
          className="btnName last number"
          name="."
          onClick={[Function]}
          type="button"
        >
           
          .
        </button>
        <button
          className="btnName last cal"
          name="="
          onClick={[Function]}
          type="button"
        >
           
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
